### PR TITLE
Add new sphinx rtd theme release 1.2.0

### PR DESCRIPTION
**Pull request by MicK7**
This new release helps with supporting more recent version of docutils

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py [hunk 1]
```diff
@@ -23,7 +24,8 @@ class PySphinxRtdTheme(PythonPackage):
     depends_on("py-sphinx@1.6:", when="@1:", type=("build", "run"))
     depends_on("python@2.7:2.8,3.4:", when="@1:", type=("build", "run"))
     depends_on("py-docutils@:0.16", when="@0.5.2:0", type=("build", "run"))
-    depends_on("py-docutils@:0.17", when="@1:", type=("build", "run"))
+    depends_on("py-docutils@:0.17", when="@1:1.1", type=("build", "run"))
+    depends_on("py-docutils@:0.20", when="@1.2:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py]: These deps don't look correct:
```
install_requires =
        sphinx >=1.6,<7
        docutils <0.19
        sphinxcontrib-jquery >=2.0.0,!=3.0.0 ; python_version > '3'
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py (hunk 1 above)
Comment (MicK7) [reply var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py]: You are right. I tested with docutils 0.19 without any issues for my project but the official release do not support 0.19 yet.
I will put back the official supported version and also add the missing sphinxcontrib-jquery dependency.

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py [hunk 2]
```diff
@@ -23,7 +24,9 @@ class PySphinxRtdTheme(PythonPackage):
     depends_on("py-sphinx@1.6:", when="@1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py]: Latest release should be `@1.6:6`, not sure when this change was made.

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py [hunk 3]
```diff
@@ -23,7 +24,9 @@ class PySphinxRtdTheme(PythonPackage):
...
     depends_on("python@2.7:2.8,3.4:", when="@1:", type=("build", "run"))
     depends_on("py-docutils@:0.16", when="@0.5.2:0", type=("build", "run"))
-    depends_on("py-docutils@:0.17", when="@1:", type=("build", "run"))
+    depends_on("py-docutils@:0.17", when="@1:1.1", type=("build", "run"))
+    depends_on("py-docutils@:0.18", when="@1.2:", type=("build", "run"))
+    depends_on("py-sphinxcontrib-jquery@2.0:3.0", when="@1.2:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py]:
```suggestion
    depends_on("py-sphinxcontrib-jquery@2,3.0.1:", when="@1.2:", type=("build", "run"))
```
or
```suggestion
    depends_on("py-sphinxcontrib-jquery@2:", when="@1.2:", type=("build", "run"))
    conflicts("^py-sphinxcontrib-jquery@3.0.0")
```

File: var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+class PySphinxcontribJquery(PythonPackage):
+    """A sphinx extension which ensure that jQuery is installed with Sphinx."""
+
+    homepage = "https://github.com/sphinx-contrib/jquery"
+    pypi = "sphinxcontrib-jquery/sphinxcontrib-jquery-2.0.0.tar.gz"
+
+    version("2.0.0", sha256="a8fb65f6dba84bf7bcd1aea1f02ab3955ac34611d838bcc95d4983b805b234da")
+
+    depends_on("python@3.5:", when="@2:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py]:
```suggestion
    depends_on("py-setuptools@65:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on("py-flit-core", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py]: flit-core isn't used in 2.0.0

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by MicK7** [comment]
@adamjstewart Thanks for the feedback

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py [hunk 6]
```diff
@@ -20,10 +21,13 @@ class PySphinxRtdTheme(PythonPackage):
 
     depends_on("py-setuptools", type="build")
     depends_on("py-sphinx", when="@0.4.1:", type=("build", "run"))
-    depends_on("py-sphinx@1.6:", when="@1:", type=("build", "run"))
+    depends_on("py-sphinx@1.6:6", when="@1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py:24]: Technically there's no upper bound for 1.0.0 but we can keep this to play it safe

File: var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    """A sphinx extension which ensure that jQuery is installed with Sphinx."""
+
+    homepage = "https://github.com/sphinx-contrib/jquery"
+    pypi = "sphinxcontrib-jquery/sphinxcontrib-jquery-2.0.0.tar.gz"
+
+    version("2.0.0", sha256="8fb65f6dba84bf7bcd1aea1f02ab3955ac34611d838bcc95d4983b805b234daa")
+
+    depends_on("python@3.5:", when="@2:", type=("build", "run"))
+    depends_on("py-setuptools@65:", type=("build", "run"))
+    depends_on("py-flit-core", when="@3:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py]:
```suggestion
    depends_on("py-flit-core@3.7:", when="@3:", type="build")
```
Might as well add 3.0.0 at this point

Comment (MicK7) [reply var/spack/repos/builtin/packages/py-sphinxcontrib-jquery/package.py]: Well as no package is requiring 3.0.0 right now, I don't see the need to add 3.0.0 . Putting py-flit-core right now is more to stay on the safe side when someone might want to introduce 3.1 version (that is the actual version we would like to target)

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py (hunk 6 above)
Comment (MicK7) [reply var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py:24]: I agree

**Review by adamjstewart — APPROVED** [review]

**Comment by MicK7** [comment]
@adamjstewart Thanks a lot for taking the time to review this merge request.

**Pull request merged**